client/src/components/Map/UCIMap.js

```diff
@@ -45,6 +47,125 @@ export default class UCIMap extends PureComponent {
         selected_acronym: '',
         eventsInCalendar: AppStore.getEventsInCalendar(),
         currentScheduleIndex: AppStore.getCurrentScheduleIndex(),
+        poly: [],
+    };
+
+    getRoute = (day) => {
+        let index = 0;
+        let coords = '';
+        let coords_array = [];
+        let colors = [];
+        this.state.eventsInCalendar
+            .sort((event, event2) => event.start - event2.start)
+            .forEach((event) => {
+                // Filter out those in a different schedule or those not on a certain day (mon, tue, etc)
+                if (
+                    event.isCustomEvent ||
+                    !event.scheduleIndices.includes(this.state.currentScheduleIndex) ||
+                    !event.start.toString().includes(DAYS[day])
+                )
+                    return;
+
+                // Get building code, get id of building code, which will get us the building data from buildingCatalogue
+                const buildingCode = event.bldg.split(' ')[0];
```

Comment:
A few buildings have 2 word building codes ("UN CLB", "Student Center", "Mesa Court", "Middle Earth",  "UCI Health",  "Hitachi, PH", "Dean's Office").

It's unlikely that there are classes here, but it's probably better to be safe, especially if new 2-word building codes are added in the future.

```suggestion
                const buildingCode = event.bldg.split(' ').slice(0, -1).join(' ');
```